elastica/joint.py

```diff
@@ -222,51 +228,68 @@ def __init__(self, k, nu, kt):
             Damping coefficient of the joint.
         kt: float
             Rotational stiffness coefficient of the joint.
+        nut: float = 0.
+            Rotational damping coefficient of the joint.
+        use_static_rotation: bool
+            If True, the initial rotation offset between the two systems is recorded
+            and enforced throughout the entire simulation. (default=True)
         """
         super().__init__(k, nu)
         # additional in-plane constraint through restoring torque
         # stiffness of the restoring constraint -- tuned empirically
         self.kt = kt
+        self.nut = nut
+
+        # if a static rotation offset between the two systems should not be enforced,
+        # we set the relative rotation to the identity matrix. Otherwise
+        if use_static_rotation:
+            self.static_rotation = None
+        else:
+            self.static_rotation = np.eye(3)
 
     # Apply force is same as free joint
     def apply_forces(self, rod_one, index_one, rod_two, index_two):
         return super().apply_forces(rod_one, index_one, rod_two, index_two)
 
-    def apply_torques(self, rod_one, index_one, rod_two, index_two):
-        # current direction of the first element of link two
-        # also NOTE: - rod two is fixed at first element
-        link_direction = (
-            rod_two.position_collection[..., index_two + 1]
-            - rod_two.position_collection[..., index_two]
+    def apply_torques(self, system_one, index_one, system_two, index_two):
+        # collect directors of systems one and two
+        # note that systems can be either rods or rigid bodies
+        system_one_director = system_one.director_collection[..., index_one]
+        system_two_director = system_two.director_collection[..., index_two]
+
+        if self.static_rotation is None:
+            # this if clause should be active during the first timestep for the case use_static_rotation==True
+            self.static_rotation = system_one_director @ system_two_director.T
```

Comment:
won't be needed if these things are set in initializer.

Reply:
Fixed in https://github.com/GazzolaLab/PyElastica/pull/135/commits/cb84464402707586dbc511df29a4156d42a7bdc0